Autocompleter+CoreDataClass.swift

```diff
@@ -0,0 +1,7 @@
+import Foundation
+import CoreData
+
+@objc(Autocompleter)
+public class Autocompleter: NSManagedObject {
```

Comment:
Do you think we really need this entity? `Autocompleter` doesn't seem like an entity, it looks to me that we can add the `siteID` as a relationship in `UserAutocomplete` and things are going to work the same way. Wdyt?

Reply:
Originally I was going to just use `UserAutocomplete` with the `siteID` as a property, but that wouldn't allow the app to differentiate between a scenario where a site's mentions were fetched from the network and there were none (empty array), and a scenario where we haven't yet fetched a site's mentions. 

This would be analogous to an optional property, e.g. an a property of type array that can be either `nil`, an empty array (or an array of values). I think the only way to model this in Core Data is with an extra entity, so I created `Autocompleter` (and moved the `siteID` there). Now, if there's no `Autocompleter` for a particular `siteID`, or there is but its `UserAutocomplete` is `nil`, the app knows that it hasn't yet fetched mentions for that site.

Soon, once we implement x-posting in Gutenberg, my idea here was to add another entity (e.g. `XPostAutocomplete`) which will also have a relationship to `Autocomplete`. (Or possibly refactor `UserAutocomplete` to support x-posting, if the data we need to store is generic enough.)

Does this sound like a good approach in your opinion?

Reply:
Hmm, making additional calls would be a problem here?

In the first solution, we had a session cache-based, maybe we can implement a similar logic here and keep the database unaware of it? If the user closes the app and reopens again, it will make a new request, but I don't think that's a problem.

My main reasoning here is to try to remove this kind of knowledge from the database layer (the database should only model the objects).

I'm not sure I fully understood the x-posting. This will autocomplete based on blog URLs?

Reply:
My latest commits remove this entity, you were right about this not being needed for the reasons I shared above. An optional 1-to-many relationship from `Blog` to `AtMentionSuggestion` allows me to differentiate between the app not having fetched suggestions yet, and having fetched but getting an empty list.

The only drawback of my new approach is: imagine a user (e.g. an A11n) who has 5 sites each with 1000 @-mentions suggestions for each site would result in 5000 rows in the `AtMentionSuggestion` entity table. What's important to keep in mind though is that `AtMentionSuggestion` is cleared when the user closes their session (it's wired to cascade delete when the `Blog` are deleted, which happens on logout.)